src/middleware/userMiddleware.js

```diff
@@ -1,30 +1,25 @@
-const { isValidString } = require("../utilities/stringUtilities");
+const { isValidBodyProperty } = require("../utilities/routerUtilities");
 
 function validateUsername(req, res, next) {
-    const username = req.body.username;
-
-    if(!isValidString(username)) {
-        res.status(400).json({
-            message: "Invalid username"
-        });
-        return;
+    if(isValidBodyProperty(req, res, "username")) {
+        next();
     }
-    next();
 }
 
 function validatePassword(req, res, next) {
-    const password = req.body.password;
+    if(isValidBodyProperty(req, res, "password")) {
+        next();
+    }
+}
 
-    if(!isValidString(password)) {
-        res.status(400).json({
-            message: "Invalid password"
-        });
-        return;
+function validateRole(req, res, next) {
```

Comment:
The validateRole here specifically is what I am talking about.

Reply:
Currently, the service layer only allows you to promote users to admin, so any role other than "admin" would fail. You're right that it should still check for a valid role, so I'll add to `userMiddleware`.